geode-core/src/main/java/org/apache/geode/internal/cache/BucketRegion.java

```diff
@@ -1558,6 +1558,7 @@ protected void distributedRegionCleanup(RegionEventImpl event) {
     // No need to close advisor, assume its already closed
     // However we need to remove our listener from the advisor (see bug 43950).
     this.distAdvisor.removeMembershipListener(this.advisorListener);
+    super.distributedRegionCleanup(event);
```

Comment:
This looks dangerous. At the very least you should get rid of the comment that says "No need to close advisor" since super.distributedRegionCleanup does that. I'm not sure, but I think if you do the super call then you do not need to call removeMembershipListener first. I think the impl in super may take care of that. In that case no need to override on BucketRegion. But I'm concerned that this make cleanup some shared resource that other buckets still needs.

Reply:
This change is removed from the PR. This will be addressed as part of separate ticket.